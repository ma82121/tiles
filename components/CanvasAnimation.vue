<template>
  <canvas v-show="play" ref="canvas" class="canvas" @click="click" />
</template>

<script>
// canvas-animation
// init, play, stop, loop
import Confetti from '@/assets/scripts/canvas-animation/confetti/app'

export default {
  name: 'CanvasAnimation',
  props: {
    play: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {
      screen: {
        width: 0,
        height: 0,
        centerX: 0,
        centerY: 0
      },
      animationName: '',
      confetti: new Confetti()
    }
  },
  watch: {
    play(bool) {
      if (bool) {
        this.start()
      }
    }
  },
  mounted() {
    window.addEventListener('resize', this.resizeCanvas, false)
    this.resizeCanvas()
    if (this.play) {
      this.start()
    }
  },
  destroyed() {
    window.removeEventListener('resize', this.resizeCanvas, false)
  },
  methods: {
    click(e) {
      this.$emit('click', e)
    },
    resizeCanvas() {
      this.$refs.canvas.style.width = window.innerWidth + 'px'
      this.$refs.canvas.style.height = window.innerHeight + 'px'
      this.$refs.canvas.width = window.innerWidth * window.devicePixelRatio
      this.$refs.canvas.height = window.innerHeight * window.devicePixelRatio
      this.screen.width = window.innerWidth * window.devicePixelRatio
      this.screen.height = window.innerHeight * window.devicePixelRatio
      this.screen.centerX = (window.innerWidth * window.devicePixelRatio) / 2
      this.screen.centerY = (window.innerHeight * window.devicePixelRatio) / 2
    },
    start() {
      this.confetti.init(this.$refs.canvas)
      this.loop()
    },
    loop() {
      if (this.play) {
        this.confetti.loop()
        window.requestAnimationFrame(this.loop)
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.canvas {
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 20;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
}
</style>
