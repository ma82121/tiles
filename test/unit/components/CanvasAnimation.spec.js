import { shallowMount } from '@vue/test-utils'
import CanvasAnimation from '@/components/CanvasAnimation'

describe('CanvasAnimation', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(CanvasAnimation)
    expect(wrapper.element).toMatchSnapshot()
  })
})
