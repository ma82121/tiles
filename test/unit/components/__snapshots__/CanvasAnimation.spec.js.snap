// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasAnimation snapshot 1`] = `
<canvas
  class="canvas"
  height="768"
  style="display: none; width: 1024px; height: 768px;"
  width="1024"
/>
`;
